src/mempool.rs

```diff
@@ -116,21 +117,28 @@ impl Mempool {
         for txid in to_remove {
             self.remove_entry(txid);
         }
-        let entries: Vec<_> = to_add
-            .par_iter()
-            .filter_map(|txid| {
-                match (
-                    daemon.get_transaction(txid, None),
-                    daemon.get_mempool_entry(txid),
-                ) {
-                    (Ok(tx), Ok(entry)) => Some((txid, tx, entry)),
-                    _ => None,
-                }
-            })
-            .collect();
-        let added = entries.len();
-        for (txid, tx, entry) in entries {
-            self.add_entry(*txid, tx, entry);
+        let to_add: Vec<Txid> = to_add.into_iter().collect();
+        let mut added = 0;
+        for chunk in to_add.chunks(100) {
+            if exit_flag.poll().is_err() {
+                warn!("interrupted while syncing mempool");
```

Comment:
Sorry for late comment but maybe this should be `info`? IIUC there's nothing wrong/dangerous about it.

Reply:
Good catch, thanks!